fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryRecursivePlan.java

```diff
@@ -0,0 +1,299 @@
+/*
+ * RecordQueryRecursivePlan.java
```

Comment:
Can you straighten out the namings please?

- logical expression -> `RecursiveUnionExpression`
- strategy is called: `level`
  - physical: `RecordQueryRecursiveUnionPlan`
  - implementor: `ImplementRecursiveUnionPlan`
- strategy is called: preorder
  - physical: `RecordQueryRecursivePlan`
  - implementor: `ImplementRecursiveDfsPlan`

Can we be consistent on the naming? Maybe something like:
- logical expression -> `RecursiveUnionExpression` (as is)
- strategy is called: `level`
  - physical: `RecordQueryRecursiveLevelUnionPlan`
  - implementor: `ImplementRecursiveLevelUnionPlan`
- strategy is called: preorder
  - physical: `RecordQueryRecursivePreorderUnionPlan`
  - implementor: `ImplementRecursivePreorderUnionPlan`